src/WebJobs.Script/Workers/WorkerConcurrencyManager.cs

```diff
@@ -125,8 +128,17 @@ internal async void OnTimer(object sender, System.Timers.ElapsedEventArgs e)
 
                 if (NewWorkerIsRequired(workerStatuses, _addWorkerStopwatch.GetElapsedTime()))
                 {
-                    await _functionInvocationDispatcher.StartWorkerChannel();
-                    _addWorkerStopwatch = ValueStopwatch.StartNew();
+                    if (_functionInvocationDispatcher is RpcFunctionInvocationDispatcher rpcDispatcher)
+                    {
+                        // Check memory
+                        if (IsEnoughMemory(Process.GetCurrentProcess().PrivateMemorySize64,
```

Comment:
Would there be any perf benefit of moving the `Process.GetCurrentProcess().PrivateMemorySize64` expression outside of the Timer tick event handler? Will the value change between ticks? if not, should we get it in the StartAsync / Activate method?

Reply:
Same question for `AppServicesHostingUtility.GetMemoryLimitBytes()`